bixlabs/authentication
# [AM-17] Configure transactional email services

## Description 
The aim of this pr is to add the implementation of some email providers, like sendgrid, mailgun and smtp.

## Stories
https://bixlabs.atlassian.net/browse/AM-17

## List of changes 
* add: mailgun implementation
* add: sendgrid implementation
* add: smtp implementation
* add: a provider initialization using env variables
* add: AuthSender struct the builds the different kinds of email messages and it sends them using a concrete sender
* add: TemplateBuilder to load and build email templates
* update: Forgot email test to check if the email message is correctly built

## Steps to Test or Reproduce
* update the cli for a quick test
     * Replace all the `email@bixlabs.com` by your email

* configure an email provider in the `.env` file

Change it to a concrete provider: (SMTP)

```bash
EMAIL_PROVIDER=smtp
SMTP_USERNAME=...
SMTP_PASSWORD=...
```

or (Mailgun)

```bash
EMAIL_PROVIDER=mailgun
MAILGUN_DOMAIN=...
MAILGUN_API_KEY=...
```

or (Sendgrid)

```bash
EMAIL_PROVIDER=sendgrid
SENDGRID_API_KEY=...
```

* Run the cli file
`make run-cli`

You should receive a forgot email message.

## Impacted Areas in Application 
* Sender
* Forgot password
